### DISCUSSION TITLE
macro support in the post JSON body

### DISCUSSION THREAD

**Discussion by AlexTM** [grafana/grafana-json-datasource#77] [Ideas]
I am using an API where all query information is submitted in a JSON blob in the body of a get or post request.

Variables are being insert correctly, however I do not think there is support for the macros $__unixEpochFrom()  and $__unixEpochTo().

Is this something that could be made available?

**Comment by marcusolsson** [comment]
Sure! I've added it now. Will probably publish another release in the coming days with the fix.

**Comment by sroul** [comment]
Hi @marcusolsson,

Could you please help me how i can use the Macros in my Grafana Dashboard Context.

Thanks
Sroul

**Reply by marcusolsson** [reply]
Add the macro string where you'd use any other dashboard variable and it will be expanded whenever you run query. Could you let me know how I can improve the documentation for [Macros](https://marcus.se.net/grafana-json-datasource/macros)?
